library/Icingadb/Data/PivotTable.php

```diff
@@ -0,0 +1,410 @@
+<?php
+
+namespace Icinga\Module\Icingadb\Data;
+
+use Icinga\Application\Icinga;
+use ipl\Orm\Query;
+use ipl\Stdlib\Contract\Paginatable;
+use ipl\Stdlib\Filter;
+
+class PivotTable
+{
+    /**
+     * The query to fetch as pivot table
+     *
+     * @var Query
+     */
+    protected $baseQuery;
+
+    /**
+     * X-axis pivot column
+     *
+     * @var string
+     */
+    protected $xAxisColumn;
+
+    /**
+     * Y-axis pivot column
+     *
+     * @var string
+     */
+    protected $yAxisColumn;
+
+    /**
+     * The filter being applied on the query for the x-axis
+     *
+     * @var Filter\Rule
+     */
+    protected $xAxisFilter;
+
+    /**
+     * The filter being applied on the query for the y-axis
+     *
+     * @var Filter\Rule
+     */
+    protected $yAxisFilter;
+
+    /**
+     * The query to fetch the leading x-axis rows and their headers
+     *
+     * @var Query
+     */
+    protected $xAxisQuery;
+
+    /**
+     * The query to fetch the leading y-axis rows and their headers
+     *
+     * @var Query
+     */
+    protected $yAxisQuery;
+
+    /**
+     * X-axis header column
+     *
+     * @var string|null
+     */
+    protected $xAxisHeader;
+
+    /**
+     * Y-axis header column
+     *
+     * @var string|null
+     */
+    protected $yAxisHeader;
+
+    protected $gridcols = [];
```

Comment:
After I've seen what you use this for, I think it's not necessary. SQL accepts aliases as alternative in most places, so the `PivotData` class shouldn't need to know the real column name I believe.

Please try to avoid this and remove it if successful. If not successful, please tell me the reason.